middleware/rabbitmq/detectors-rabbitmq.tf

```diff
@@ -0,0 +1,256 @@
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+  }
+}
+
+resource "signalfx_detector" "sockets" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] RabbitMQ Node sockets usage"
+
+  program_text = <<-EOF
+        A = data('gauge.node.sockets_used', filter=filter('plugin', 'rabbitmq') and ${module.filter-tags.filter_custom})${var.sockets_aggregation_function}
+        B = data('gauge.node.sockets_total', filter=filter('plugin', 'rabbitmq') and ${module.filter-tags.filter_custom})${var.sockets_aggregation_function}
+        signal = (A/B).scale(100).${var.sockets_transformation_function}(over='${var.sockets_transformation_window}').publish('signal')
+        detect(when(signal > ${var.sockets_threshold_critical})).publish('CRIT')
+        detect(when(signal > ${var.sockets_threshold_warning})).publish('WARN')
+  EOF
+
+  rule {
+    description           = "is too high > ${var.sockets_threshold_critical}"
+    severity              = "Critical"
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.sockets_disabled_critical, var.sockets_disabled, var.detectors_disabled)
+    notifications         = coalescelist(var.sockets_notifications_critical, var.sockets_notifications, var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+  }
+
+  rule {
+    description           = "is too high > ${var.sockets_threshold_warning}"
+    severity              = "Warning"
+    detect_label          = "WARN"
+    disabled              = coalesce(var.sockets_disabled_warning, var.sockets_disabled, var.detectors_disabled)
+    notifications         = coalescelist(var.sockets_notifications_warning, var.sockets_notifications, var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+  }
+}
+
+resource "signalfx_detector" "vm_memory" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] RabbitMQ Node vm_memory usage"
+
+  program_text = <<-EOF
+        A = data('gauge.node.mem_used', filter=filter('plugin', 'rabbitmq') and ${module.filter-tags.filter_custom})${var.vm_memory_aggregation_function}
+        B = data('gauge.node.mem_limit', filter=filter('plugin', 'rabbitmq') and ${module.filter-tags.filter_custom})${var.vm_memory_aggregation_function}
+        signal = (A/B).scale(100).${var.vm_memory_transformation_function}(over='${var.vm_memory_transformation_window}').publish('signal')
+        detect(when(signal > ${var.vm_memory_threshold_critical})).publish('CRIT')
+        detect(when(signal > ${var.vm_memory_threshold_warning})).publish('WARN')
+  EOF
+
+  rule {
+    description           = "is approaching the watermark"
```

Comment:
watermark -> limit?

Reply:
Yes, this is the name in RabbitMQ configuration (`vm_memory_high_watermark`) : https://www.rabbitmq.com/memory.html

Reply:
I don't say it is not correct but it seems more understandable in my opinion and this is just as relevant as watermark while the metric is named `gauge.node.mem_limit`